src/Model/SourcesRepository.php

```diff
@@ -79,20 +79,22 @@ public function getByOrderId(string $orderId): SourcesInterface
      * @param string $orderId
      * @param string $itemSku
      *
-     * @return SourceSelectionItem
+     * @return SourceSelectionItem []
      * @throws NoSuchEntityException
      */
-    public function getSourceItemBySku(string $orderId, string $itemSku): SourceSelectionItem
+    public function getSourceItemBySku(string $orderId, string $itemSku): array
     {
         $sourceSelectionItems = $this->sourcesConverter->convertSourcesJsonToSourceSelectionItems(
             $this->getByOrderId($orderId)->getSources()
         );
 
-        if (!array_key_exists($itemSku, $sourceSelectionItems)) {
-            throw new NoSuchEntityException(__('Source model with the sku "%1" does not exist', $itemSku));
+        foreach ($sourceSelectionItems as $sourceSelectionItem) {
+            if ($sourceSelectionItem->getSku() !== $itemSku) {
+                $sourceSelectionItem->setSourceCode('default');
```

Comment:
I'm not sure `default` is actually hard coded and guaranteed to exist? I think it may be like the default store code being `default` in magento but setup scripts can readily delete that and create as `uk` for example.

Reply:
https://github.com/AmpersandHQ/magento2-disable-stock-reservation/blob/bdb5cb2d339204cd5582600f076162e2b40b48d3/src/Service/ExecuteSourceDeductionForItems.php#L140-L142C14
Yes, this part confused me abit.
From what i can see the error thrown is caught is this class `ExecuteSourceDeductionForItems.php` and the sourceCode hardcoded to 'default' and this is why i  made that change.

Reply:
@convenient the default sourcecode actually exists
```
mysql> mysql> select * from inventory_source_item where sku ='64411d69-0361-49f7-bb85-4a30bb75bbeb';
+----------------+-------------+--------------------------------------+----------+--------+
| source_item_id | source_code | sku                                  | quantity | status |
+----------------+-------------+--------------------------------------+----------+--------+
|       10241017 | BE          | 64411d69-0361-49f7-bb85-4a30bb75bbeb |   2.0000 |      1 |
|       10201427 | default     | 64411d69-0361-49f7-bb85-4a30bb75bbeb |   2.0000 |      1 |
|       10232247 | NL          | 64411d69-0361-49f7-bb85-4a30bb75bbeb |   2.0000 |      1 |
+----------------+-------------+--------------------------------------+----------+--------+
```

Reply:
https://experienceleague.adobe.com/docs/commerce-knowledge-base/kb/troubleshooting/miscellaneous/cannot-delete-the-source-or-change-its-code.html?lang=en

It should be possible for me to decide to rename or remove `default` to be `uk` in this case for example. For our specific site it does exist, but this module plugs into many sites of many configurations.

Not sure we can rely on hardcoded string `default` for this handling.

Reply:
@convenient default stock/source cannot be removed/renamed. It is possible  to rename ore remove other sources but not the default source.
https://experienceleague.adobe.com/docs/commerce-admin/inventory/stocks/stocks-manage.html
```
You start with a Default Stock that cannot be removed or disabled. 
You can add additional sales channels to the stock only. The only assigned source is Default Source. 
This stock is used by single-source Merchants, third-party integrations, and imported products.
```

Reply:
This use case happens when you assign multiple stock to a product, then unassign it after placing an order. Because the stock is not assigned to the product, the stock is returned to the default stock

Reply:
> default stock/source cannot be removed/renamed

Amazing thanks for confirming that :)